plugins/org.eclipse.n4js/src/org/eclipse/n4js/typesystem/ExpectedTypeJudgment.java

```diff
@@ -287,6 +290,29 @@ public TypeRef casePostfixExpression(PostfixExpression e) {
 
 		@Override
 		public TypeRef caseUnaryExpression(UnaryExpression e) {
+			Expression innerExpression = e.getExpression();
+			TypeRef typeRef = ts.tau(innerExpression);
+
+			if (innerExpression instanceof UnaryExpression) {
+				typeRef = ts.tau(((UnaryExpression) innerExpression).getExpression());
+			}
+
+			boolean isNumber = RuleEnvironmentExtensions.isNumeric(G, typeRef);
+
+			EObject parent = e.eContainer();
+			if (parent instanceof UnaryExpression) {
+				parent = ((UnaryExpression) parent).eContainer();
+			}
+
+			if (parent instanceof RelationalExpression) {
+				RelationalExpression re = (RelationalExpression) parent;
+				boolean isInstanceOf = re.getOp().equals(RelationalOperator.INSTANCEOF);
+
+				if (isInstanceOf) {
```

Comment:
Could you add a comment here why we use "anyTypeRef" here -- e.g. reference AbstractN4JSDeclarativeValidator.checkRelationalExpression